changelogs/fragments/5629-add-prepend-hash-option-for-channel-id.yml

```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - "slack - fix message update for channels which start with a prefix not in the hardcoded list, for example ``G4``. When ``message-id`` was passed it failed for channels which started with ``G4`` because the ``#`` symbol was added before the ``channel_id``. Adding a prepend_hash option to workaround this issue for channel id prefixes not in the list. There are three-value options:``prepend_hash: always | never | auto``, with current default auto to maintain current behavior (https://github.com/ansible-collections/community.general/pull/5629)."
```

Comment:
```suggestion
  - "slack - add option ``prepend_hash`` which allows to control whether a ``#`` is prepended to ``channel_id``. The current behavior (value ``auto``) is to prepend ``#`` unless some specific prefixes are found. That list of prefixes is incomplete, and there does not seem to exist a documented condition on when exactly ``#`` must not be prepended. We recommend to explicitly set ``prepend_hash=always`` or ``prepend_hash=never`` to avoid any ambiguity (https://github.com/ansible-collections/community.general/pull/5629)."
```